scripts/python/check_jbod_psu_status.py

```diff
@@ -0,0 +1,87 @@
+from integralstor import command, alerts
+import re, sys
+
+def get_enclosure_devices():
+    # Returns a list of /dev/sgX path for Connected Storage Enclosures
+    enclosures = []
+    try:
+        sg_devs, err = command.get_command_output('sg_map -i')
+        if err:
+            raise Exception(err)
+
+        if sg_devs:
+            for sg_dev in sg_devs:
+                    encl = re.findall('ENCL', sg_dev, re.IGNORECASE)
+                    if encl:
+                        res = re.findall(r'^\/\w+\/\w+', sg_dev)
+                        enclosures += res
+    except Exception, e:
+        return None, "Error retrieving enclosures : %s" % str(e)
+    else:
+        return enclosures, None
+
+def get_psu_status():
+    psu_status = {}
+    try:
+        encls, err = get_enclosure_devices()
+        if err:
+            raise Exception(err)
+
+        if encls:
+            for encl in encls:
+                for psu in range(2):
+                    tmp, err = command.get_command_output('sg_ses --index=ps,%d %s' % (psu, encl))
+                    if err:
+                        raise Exception(err)
+                    if tmp:
+                        for status in tmp:
+                            result = re.findall(r'status: \w+', status)
+                            if result:
+                                psu_status['PSU-%d' % psu] = result[0]
```

Comment:
Since this returns the status of all the psus, should psu_status be a list rather than a dictionary and its index can refer to the psu number. That way, it is easier to just iterate through all the PSUs for status.

Reply:
Cld you pls look into this one also?